File: project-Euler/javascript/3.js
function largestPrimeFactor(number){
let lv = 0;

  while (number % 2 == 0) {
    lv = 2;
    number = Math.floor(number / 2);
  }

  for (let i = 3; i <= Math.sqrt(number); i += 2) {
    while (number % i == 0) {
      lv = i;
      number = Math.floor(number / i);
    }
  }

  if (number > 2) {
    lv = number;
  }

  return lv;
};

console.log(largestPrimeFactor(2))

File: project-Euler/python/3.py
import math
def largestPrimeFactor(num):
    lv = 0
    while num%2 == 0:
        lv = 2 
        num = num//2
    for i in range(3,math.floor(math.sqrt(num)),2):
        while num % i == 0:
            lv = i
            num = num//i
    if num>2:
        lv =num
    return lv

print(largestPrimeFactor(600851475143))